app/assets/javascripts/startpage/progressBar.coffee

```diff
@@ -0,0 +1,52 @@
+progressBar = angular.module("ProgressBar", ["StartPageServices"])
+
+progressBar.config(["$routeProvider", ($routeProvider) ->
+    $routeProvider.when("/progress_bar", {
+        templateUrl: "/assets/progress_bar.html",
+        controller: "ProgresBarController"
+    })
+])
+
+progressBar.controller("ProgressBarController", ["$scope", "$http", "$location", "$routeParams", "SharedRequests", "StartPageData", ($scope, $http, $location, $routeParams, SharedRequests, StartPageData) ->
+
+    currentID = StartPageData.getCurrentTopic()
+    percentComplete = 0
+
+    $scope.numQuestions = () -> StartPageData.getNumQuestions()
+    $scope.questionsLeft = () -> StartPageData.getQuestionsLeft()
+    $scope.numTopics = () -> StartPageData.getNumTopics()
+    $scope.isTopicDone = () -> StartPageData.isTopicQuestionsDone(currentID)
+
+    $scope.testvar = () -> return "50%"
+
+    # %complete = (numCompleteTopics * 100/numTopics) + (numAnsweredQuestions * 100/totalQuestions * 100/numTopics)
+    # calling $scope vs calling StartPageData ???
+    $scope.percentComplete = () ->
+        $scope.updateCompleteTopics() # update complete topics first
+
+        numAnsweredQuestions = $scope.numQuestions() - $scope.questionsLeft()
+        numCompleteTopics = StartPageData.getTopicsComplete()
+
+        pastPercent = numCompleteTopics * 100 / $scope.numTopics()
+        currentPercent = numAnsweredQuestions * 100 / $scope.numQuestions() / $scope.numTopics()
+
+        percentComplete = pastPercent + currentPercent
+
+        if percentComplete > 100 # might need to change this later because it might break stuff
+            percentComplete = 100
+
+        return percentComplete
+
+    $scope.percentCompleteString = () -> # for ng-style
+        n = percentComplete.toString()
+        return n + "%"
+
+    # Updates the # of complete topics variable
+    $scope.updateCompleteTopics = () ->
```

Comment:
You don't need to bind this to $scope. I would make it local to the controller scope (i.e. just do updateCompleteTopics = () -> ......)